controllers/photos.js

```diff
@@ -0,0 +1,140 @@
+const axios = require('axios')
+const { imageFromBuffer } = require('@canvas/image')
+const { hash } = require('./imageHashing')
+
+/**
+ * Deduplicates array of image urls by file name and image 
+ * render returning lategest images when duplicates are found
+ * @param {Array} images - array of image urls
+ * @returns {Array} - returns an array of url 
+ */
+const getUniqueImageUrls = async (imageUrls) => {
+  const uniqueImageMap = await _dedupeByFileName(imageUrls)
+  const imgMap = await _dedupeByImageRender(uniqueImageMap)
+  const uniqueImages = []
+  for (let [key, value] of imgMap) {
+    uniqueImages.push(value.url)
+  }
+  return uniqueImages
```

Comment:
Could use an `array.map` here instead, but unfortunately you need to handle the Map since it does not support `.map`.

```suggestion
  return Array.from(imgMap.values()).map(value => value.url)
```